### DISCUSSION TITLE
0082 XLS-82d: Multi-Purpose Tokens (MPTs) integration into XRP Ledger DEX

### DISCUSSION THREAD

**Discussion by gregtatcam** [XRPLF/XRPL-Standards#231] [XLS Proposals]
## The earlier version can be found by looking at the edit history.

**Comment by donovanhide** [comment]
My preference would be strongly for the blackhole account flag for serializing the mpt_issuance_id. Having a white/blacklist of non-standard currencies is ugly and would add a hash map lookup for every Offer and AMM ledger entry in whole ledger operations. 

typo:
s/red/read/

"then additional 32 bits have to be __red__ and deserialized into a sequence number."

**Comment by melitus** [comment]
Can one use MFT to implement fractional NFT?

**Reply by kennyzlei** [reply]
Yes, you can use MPT to implement fractional NFT.  You can use the `MaximumAmount` field in `MPTokenIssuance` transaction to indicate the number of fractional units of an NFT, where an account holding all units will have 100% ownership

https://xrpl.org/docs/references/protocol/ledger-data/ledger-entry-types/mptokenissuance

**Comment by Island-Ghost** [comment]
Can you migrate old tokens to the MPT?

**Reply by gregtatcam** [reply]
Not at the protocol level. A client application should be able to do this.

**Comment by Tapanito** [comment]
@gregtatcam what is the status of this specification? Can it be moved to a pull-request?

**Reply by gregtatcam** [reply]
I'd like to make sure that implementation and the specs is in sync. Hopefully I can do this in the next week or so.

**Comment by Wawa-Meridian** [comment]
Is there any plan to move this up for a formal vote?
